### PR TITLE
feat: add validation rule for charge owner text length

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Length of charge owner must be between 13 (GLN) and 16 (EIC)

## References

#1326

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeOwnerTextLengthRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace GreenEnergyHub.Charges.Tests.Domain.Dtos.ChargeCommands.Validation.InputValidation.ValidationRules
+{
+    [UnitTest]
+    public class ChargeOwnerTextLengthRuleTests
+    {
+        [Theory]
+        [InlineAutoMoqData("Short   name", false, "the length is less than 13 characters.")]
+        [InlineAutoMoqData("Too     long name", false, "the length is longer than 13 characters.")]
+        [InlineAutoMoqData("Charge  Owner", true, "13 character GLN names are supported.")]
+        [InlineAutoMoqData("Charge Fee Name", true, "16 character EIC names are supported.")]
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ValidationRules/ChargeOwnerTextLengthRuleTests.cs:31]:
```suggestion
        [InlineAutoMoqData("Too     long name", false, "the length is longer than 16 characters.")]
        [InlineAutoMoqData("1234567890123", true, "13 character GLN names are supported.")]
        [InlineAutoMoqData("ThisIsA16charEic", true, "16 character EIC names are supported.")]
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs [hunk 2]
```diff
@@ -169,6 +169,10 @@ public static class CimValidationErrorTextTemplateMessages
         public const string ResolutionRequiredErrorText =
             "Resolution is missing.";
 
+        [ErrorMessageFor(ValidationRuleIdentifier.ChargeOwnerHasLengthLimits)]
+        public const string ChargeOwnerTextLengthRuleText =
+            "When sending a charge information (D18) the Charge Owner must have a lenght between 13 and 16.";
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs]: Changed in 7a1221deccc039f34be6530e765046c06b66506c

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs [hunk 3]
```diff
@@ -169,6 +169,10 @@ public static class CimValidationErrorTextTemplateMessages
         public const string ResolutionRequiredErrorText =
             "Resolution is missing.";
 
+        [ErrorMessageFor(ValidationRuleIdentifier.ChargeOwnerHasLengthLimits)]
+        public const string ChargeOwnerTextLengthRuleText =
+            "When sending a charge (D08 or D18) the Charge Owner must have a lenght between 13 and 16.";
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs:174]:
```suggestion
            "Charge owner must have a length between 13 and 16.";
```

**Review by prtandrup — APPROVED** [review]
LGTM with a nit comment I just commented a moment ago

**Pull request merged**